test/torchaudio_unittest/datasets/tedlium_test.py

```diff
@@ -0,0 +1,93 @@
+import os
+
+from torchaudio.datasets import tedlium
+
+from torchaudio_unittest.common_utils import (
+    TempDirMixin,
+    TorchaudioTestCase,
+    get_whitenoise,
+    save_wav,
+    normalize_wav,
+)
+
+# Used to generate a unique utterance for each dummy audio file
+UTTERANCES = [
+    "AaronHuey_2010X 1 AaronHuey_2010X 0.0 2.0 <o,f0,female> script1\n",
+    "AaronHuey_2010X 1 AaronHuey_2010X 2.0 4.0 <o,f0,female> script2\n",
+    "AaronHuey_2010X 1 AaronHuey_2010X 4.0 6.0 <o,f0,female> script3\n",
+    "AaronHuey_2010X 1 AaronHuey_2010X 6.0 8.0 <o,f0,female> script4\n",
+    "AaronHuey_2010X 1 AaronHuey_2010X 8.0 10.0 <o,f0,female> script5\n",
+]
+
+
+class TestTedlium(TempDirMixin, TorchaudioTestCase):
+    backend = "default"
+
+    root_dir = None
+    samples = []
+
+    @classmethod
+    def setUpClass(cls):
+        cls.root_dir = cls.get_base_temp_dir()
+        cls.root_dir = dataset_dir = os.path.join(cls.root_dir, "tedlium")
+        os.makedirs(dataset_dir, exist_ok=True)
+        sample_rate = 16000  # 16kHz
+        seed = 0
+        data = get_whitenoise(sample_rate=sample_rate, duration=10.00, n_channels=1, dtype="float32", seed=seed)
```

Comment:
Mocked audio data has to be generated for each files being mocked, with different seed value, otherwise all the files generated are exactly same (for the sake or test reproducibility, test helper functions are implemented deterministic by default, that is with fixed seed value) and we do not know if the dataset implementation is traversing the files in expected order.

Reply:
oh I get it now. it's one file with different segments.
Can we have multiple audio files per release?
Testing for cases with multiple objects often reveal a bug, so it's a good practice.
Here, we would like to make sure that `os.listdir` function (which returns items in OS-specific way) does not change the result across OSs. (though our test is broken for non-linux environment, which I have to fix soon.)

Reply:
Just pushed a new version with the previous feedback and including a separate test for every release. More test never hurt and as you say we could have miss some failures only testing for the 1st release.